recipes-support/blink/blink.bb

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2021 Cem Tenruh <c.tenruh@phytec.de>
+# Released under the MIT license (see COPYING.MIT for the terms)
+
+SUMMARY = "LED blink test"
+HOMEPAGE = ""
+LICENSE = "CLOSED"
+SECTION = ""
+DEPENDS = ""
+RDEPENDS_${PN} += "bash"
+
+SRC_URI = " \
+    file://blink.service \
+"
+
+SYSTEMD_SERVICE_${PN} = "blink.service"
+
+python do_write_script() {
+    blink_color = "red"
+    work_path = d.expand("${WORKDIR}")
+    blink = open('%s/blink.sh' % work_path,'w')
+
+    if blink_color == "red":
+        blink.write('''#!/bin/bash
+
+echo 0 > /sys/class/leds/led-green/brightness
+echo 0 > /sys/class/leds/led-blue/brightness
+
+echo heartbeat > /sys/class/leds/led-red/trigger''')
+
+    else:
+        blink.write('''#!/bin/bash
+
+echo 0 > /sys/class/leds/led-green/brightness
+echo 0 > /sys/class/leds/led-red/brightness
+
+echo heartbeat > /sys/class/leds/led-blue/trigger''')
+
+
+    blink.close()
+}
+
+
+addtask write_script before do_install
+
+do_install() {
+
+    install -d ${D}/usr/bin/
+    install -m 0755 ${WORKDIR}/blink.sh ${D}/usr/bin/
```

Comment:
If installing to `/usr/bin` you may want to omit the file extension `.sh` so it is easier to execute the program.